controllers/datadogagent/feature/logcollection/feature.go

```diff
@@ -0,0 +1,200 @@
+			f.tempStoragePath = *dda.Spec.Features.LogCollection.TempStoragePath
+		}
+		if dda.Spec.Features.LogCollection.OpenFilesLimit != nil {
+			f.openFilesLimit = *dda.Spec.Features.LogCollection.OpenFilesLimit
+		} else {
+			f.openFilesLimit = 100
+		}
+	}
+
+	return feature.RequiredComponents{
+		Agent: feature.RequiredComponent{Required: &f.enable},
+	}
+}
+
+// ConfigureV1 use to configure the feature from a v1alpha1.DatadogAgent instance.
+func (f *logCollectionFeature) ConfigureV1(dda *v1alpha1.DatadogAgent) feature.RequiredComponents {
+	f.owner = dda
+
+	if dda.Spec.Features.LogCollection != nil {
+		if apiutils.BoolValue(dda.Spec.Features.LogCollection.Enabled) {
+			f.enable = true
+
+			if apiutils.BoolValue(dda.Spec.Features.LogCollection.LogsConfigContainerCollectAll) {
+				f.containerCollectAll = true
+			}
+			if dda.Spec.Features.LogCollection.ContainerCollectUsingFiles == nil || *dda.Spec.Features.LogCollection.ContainerCollectUsingFiles {
+				f.containerCollectUsingFiles = true
+			}
+			if dda.Spec.Features.LogCollection.ContainerLogsPath != nil {
+				f.containerLogsPath = *dda.Spec.Features.LogCollection.ContainerLogsPath
+			}
+			if dda.Spec.Features.LogCollection.PodLogsPath != nil {
+				f.podLogsPath = *dda.Spec.Features.LogCollection.PodLogsPath
+			}
+			if dda.Spec.Features.LogCollection.ContainerSymlinksPath != nil {
+				f.containerSymlinksPath = *dda.Spec.Features.LogCollection.ContainerSymlinksPath
+			}
+			if dda.Spec.Features.LogCollection.TempStoragePath != nil {
+				f.tempStoragePath = *dda.Spec.Features.LogCollection.TempStoragePath
+			}
+			if dda.Spec.Features.LogCollection.OpenFilesLimit != nil {
+				f.openFilesLimit = *dda.Spec.Features.LogCollection.OpenFilesLimit
+			} else {
+				f.openFilesLimit = 100
+			}
+		}
+	}
+
+	return feature.RequiredComponents{
+		Agent: feature.RequiredComponent{Required: &f.enable},
+	}
+}
+
+// ManageDependencies allows a feature to manage its dependencies.
+// Feature's dependencies should be added in the store.
+func (f *logCollectionFeature) ManageDependencies(managers feature.ResourceManagers) error {
+	return nil
+}
+
+// ManageClusterAgent allows a feature to configure the ClusterAgent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *logCollectionFeature) ManageClusterAgent(managers feature.PodTemplateManagers) error {
+	return nil
+}
+
+// ManageNodeAgent allows a feature to configure the Node Agent's corev1.PodTemplateSpec
+// It should do nothing if the feature doesn't need to configure it.
+func (f *logCollectionFeature) ManageNodeAgent(managers feature.PodTemplateManagers) error {
+	// pointerdir volume mount
+	var pointerVol, logPodVol, logContainerVol, symlinkVol corev1.Volume
+	var pointerVolMount, logPodVolMount, logContainerVolMount, symlinkVolMount corev1.VolumeMount
+	// TODO: mount as ReadWrite instead of ReadOnly
+	if f.tempStoragePath != "" {
```

Comment:
i think this check isn't needed because it will be set above, unless i'm missing something?